pkg/minikube/update/update.go

```diff
@@ -129,91 +164,68 @@ func getLatestVersionFromGitHub(githubOwner, githubRepo string) (semver.Version,
 	return semver.Version{}, fmt.Errorf("Cannot get release name.")
 }
 
-func writeTimeToFile(path string, inputTime time.Time) error {
-	err := ioutil.WriteFile(path, []byte(inputTime.Format(timeLayout)), 0644)
-	if err != nil {
-		return fmt.Errorf("Error writing current update time to file: %s", err)
-	}
-	return nil
-}
-
-func getTimeFromFileIfExists(path string) time.Time {
-	lastUpdateCheckTime, err := ioutil.ReadFile(path)
-	if err != nil {
-		return time.Time{}
-	}
-	timeInFile, err := time.Parse(timeLayout, string(lastUpdateCheckTime))
-	if err != nil {
-		return time.Time{}
-	}
-	return timeInFile
-}
-
-func updateBinary(v semver.Version, downloadBinary, updateLinkPrefix, downloadLinkFormat string) {
-	checksum, err := downloadChecksum(v, downloadBinary, downloadLinkFormat)
-	if err != nil {
-		glog.Errorf("Cannot download checksum: %s", err)
-		os.Exit(1)
-	}
+// downloadAndVerifyArchive downloads the archive of latest version of minishift from GitHub, saves it to a temporary location,
+// and verifies the checksum of downloaded archive with checksum available on GitHub.
+// It returns a string containing path to the downloaded archive.
+// It returns an error if any of the steps failed.
+func downloadAndVerifyArchive(latestVersion semver.Version, archiveName, downloadLinkFormat string) (string, error) {
+	url := fmt.Sprintf(downloadLinkFormat, latestVersion, archiveName)
```

Comment:
why don't you not just pass the url into this method? This seems to be all that is really relevant, right?